torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -117,12 +117,22 @@ def selective_checkpointing_context_fn():
 
 def get_tp_parallel_strategy(
     job_config: JobConfig,
+    model: nn.Module,
 ) -> Tuple[RowwiseParallel, ColwiseParallel, PrepareModuleInput]:
     """Get the parallel strategy for the transformer model.
 
     This function handles the special case of using float8 with tensor parallelism.
     """
-    if job_config.training.fp8_linear == "dynamic":
+    if job_config.training.enable_fp8_linear:
+        from float8_experimental.float8_linear import Float8Linear, TensorScalingType
+
+        if any(
+            isinstance(m, Float8Linear)
+            and m.scaling_type_w is TensorScalingType.DELAYED
+            for m in module.modules()
+        ):
+            raise NotImplementedError("Only supports delayed scaling")
```

Comment:
Should be "only supports dynamic scaling" right?

but also we we just landed support for this right? Maybe not tested yet with f-8 all gather?

cc @vkuzo

Reply:
good catch! will change it